coops-ui/tests/unit/AddressListSm.spec.ts

```diff
@@ -4,79 +4,132 @@ import Vuelidate from 'vuelidate'
 
 import store from '@/store/store'
 import AddressListSm from '@/components/Dashboard/AddressListSm.vue'
+import { mount } from '@vue/test-utils'
+import { EntityTypes } from '@/enums'
 
 Vue.use(Vuetify)
 Vue.use(Vuelidate)
 
 let vuetify = new Vuetify({})
 
 describe('AddressListSm', () => {
-  let vm
-
-  beforeEach(done => {
-    const Constructor = Vue.extend(AddressListSm)
-    const instance = new Constructor({ store: store, vuetify })
-    vm = instance.$mount()
-
-    Vue.nextTick(() => {
-      done()
-    })
-  })
-
   it('handles empty data', done => {
     // init store
-    store.state.mailingAddress = null
-    store.state.deliveryAddress = null
+    store.state.registeredAddress = null
+    store.state.recordsAddress = null
 
-    Vue.nextTick(() => {
-      expect(vm.mailingAddress).toBeNull()
-      expect(vm.deliveryAddress).toBeNull()
-      expect(vm.$el.querySelectorAll('.list-item').length).toEqual(0)
+    const wrapper = mount(AddressListSm, { store, vuetify })
+    const vm = wrapper.vm as any
 
+    Vue.nextTick(async () => {
+      expect(vm.registeredAddress).toBeNull()
+      expect(vm.recordsAddress).toBeNull()
+
+      wrapper.destroy()
       done()
     })
   })
 
-  it('displays the mailing address', done => {
-    // init store
-    store.state.mailingAddress = {
-      'streetAddress': '1012 Douglas St',
-      'addressCity': 'Victoria',
-      'addressRegion': 'BC',
-      'postalCode': 'V8W 2C3',
-      'addressCountry': 'CA'
+  it('displays registered office data when a Coop', done => {
+    // Init Store
+    store.state.registeredAddress = {
+      'deliveryAddress':
+        {
+          'streetAddress': '220 Buchanan St',
+          'addressCity': 'Glasgow',
+          'addressRegion': 'Scotland',
+          'postalCode': 'G1 2FFF',
+          'addressCountry': 'CA'
+        },
+      'mailingAddress':
+        {
+          'streetAddress': '220 Buchanan St',
+          'addressCity': 'Glasgow',
+          'addressRegion': 'Scotland',
+          'postalCode': 'G1 2FFF',
+          'addressCountry': 'CA'
```

Comment:
Since the 2 addresses are the same, can you please add a test (or expectation) that "Same as above" is displayed?

Please also add a test case for different addresses and making sure both display.

Reply:
Done!

Reply:
Done, thanks.